libs/math/include/math/mat3.h

```diff
@@ -289,6 +289,14 @@ class MATH_EMPTY_BASES TMat33 :
         return matrix::cof(m);
     }
 
+    /**
+     * Pre-scale a matrix m by the inverse of the largest scale factor to avoid large post-transform
+     * magnitudes in the shader. This is useful for normal transformations, to avoid large
+     * post-transform magnitudes in the shader, especially in the fragment shader, where we use
+     * medium precision.
+     */
+    static constexpr TMat33<T> prescaleForNormals(const TMat33& m) noexcept;
```

Comment:
I don't think this should be on `mat3`. It can be a helper method somewhere though.